source/MRViewer/MRMouseController.cpp

```diff
@@ -196,8 +224,25 @@ bool MouseController::preMouseUp_( MouseButton btn, int )
     return false; // so others can override mouse up even if scene control was active
 }
 
-bool MouseController::preMouseMove_( int x, int y)
+bool MouseController::preMouseMove_( int x, int y )
 {
+    if ( clickButton_ != NoButton )
+    {
+        if ( std::abs( x - downMousePos_.x ) + std::abs( y - downMousePos_.y ) > cMouseClickDist ||
+             ( std::chrono::steady_clock::now() - clickTime_ ).count() > cMouseClickNs )
+        {
+            MouseButton btn = clickButton_;
+            clickButton_ = NoButton;
+            // Moved the mouse far/long enough - replay mouse down in original position
+            currentMousePos_ = downMousePos_;
+            if ( clickPendingDown_ == btn )
+                mouseDown_( btn, clickModifiers_ );
```

Comment:
we should emit mouse down when it is really called, without suppresing or deferring it